cmd/telemeter-rhel-server/main.go

```diff
@@ -0,0 +1,395 @@
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	stdlog "log"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/go-chi/chi"
+	"github.com/go-chi/chi/middleware"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/oklog/run"
+	"github.com/openshift/telemeter/pkg/tracing"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/spf13/cobra"
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+	"go.opentelemetry.io/otel"
+	"golang.org/x/oauth2"
+
+	telemeter_http "github.com/openshift/telemeter/pkg/http"
+	"github.com/openshift/telemeter/pkg/logger"
+	"github.com/openshift/telemeter/pkg/receive"
+	"github.com/openshift/telemeter/pkg/server"
+)
+
+const desc = `
+Server for receiving Prometheus metrics through the remote_write API. Clients are authenticated 
+with mTLS.
+`
+
+func defaultOpts() *Options {
+	return &Options{
+		LimitBytes:        500 * 1024,
+		LimitReceiveBytes: receive.DefaultRequestLimit,
+		clusterIDKey:      "_id",
+		Ratelimit:         4*time.Minute + 30*time.Second,
+	}
+}
+
+func main() {
+	opt := defaultOpts()
+
+	var listen, listenInternal string
+	cmd := &cobra.Command{
+		Short:         "Proxy for Prometheus remote_write API with mTLS authentication.",
+		Long:          desc,
+		SilenceErrors: true,
+		SilenceUsage:  true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			listener, err := net.Listen("tcp", listen)
+			if err != nil {
+				return err
+			}
+			internalListener, err := net.Listen("tcp", listenInternal)
+			if err != nil {
+				return err
+			}
+
+			return opt.Run(context.Background(), listener, internalListener)
+		},
+	}
+
+	cmd.Flags().StringVar(&listen, "listen", "0.0.0.0:9003", "A host:port to listen on for upload traffic.")
+	cmd.Flags().StringVar(&listenInternal, "listen-internal", "localhost:9004", "A host:port to listen on for health and metrics.")
+
+	cmd.Flags().StringVar(&opt.TLSKeyPath, "tls-key", opt.TLSKeyPath, "Path to a private key to serve TLS for external traffic.")
+	cmd.Flags().StringVar(&opt.TLSCertificatePath, "tls-crt", opt.TLSCertificatePath, "Path to a certificate to serve TLS for external traffic.")
+	cmd.Flags().StringVar(&opt.TLSCACertificatePath, "tls-ca-crt", opt.TLSCACertificatePath, "Path to the trusted Certificate Authority of the telemeter client for mTLS.")
```

Comment:
lets remove the word "telemeter" from this help text

Reply:
done